base_user_role/models/role.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 ABF OSIELL <http://osiell.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import datetime
+import logging
+
+from openerp import api, fields, models
+
+
+_logger = logging.getLogger(__name__)
+
+
+class ResUsersRole(models.Model):
+    _name = 'res.users.role'
+    _inherits = {'res.groups': 'group_id'}
+    _description = "User role"
+
+    group_id = fields.Many2one(
+        'res.groups', required=True, ondelete='cascade',
+        readonly=True, string=u"Associated group")
+    line_ids = fields.One2many(
+        'res.users.role.line', 'role_id', string=u"Users")
+    user_ids = fields.One2many(
+        'res.users', string=u"Users", compute='_compute_user_ids')
+
+    @api.multi
+    @api.depends('line_ids')
+    def _compute_user_ids(self):
+        for role in self:
+            role.user_ids = role.line_ids.mapped('user_id')
+
+    @api.model
+    def create(self, vals):
+        new_record = super(ResUsersRole, self).create(vals)
+        new_record.update_users()
+        return new_record
+
+    @api.multi
+    def write(self, vals):
+        res = super(ResUsersRole, self).write(vals)
+        self.update_users()
+        return res
+
+    @api.multi
+    def update_users(self):
+        """Update all the users concerned by the roles identified by `ids`."""
+        users = self.mapped('user_ids')
+        if users:
```

Comment:
There shouldn't be need for this guard, mapped will just return an empty Recordset that should work with the other method just fine due to record loop